### PR TITLE
feat(hardware): Hardware integration tests

### PR DESCRIPTION
**Pull request by amitlissack**
# Overview

Not ready to be merged until we can run FW simulators in CI. Just want to get a head start.

# Changelog

- conftest for common fixtures.
- tests for device info requests
- tests for motion constraint messages
- added make target to run tests against local opentrons_sock simulation set up

# Review requests

Tested using opentrons_sock can server.

# Risk assessment
None

### CODE REVIEW

File: hardware/tests/firmware_integration/test_motion_constraints.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    MotionConstraintsPayload,
+    EmptyPayload,
+)
+from opentrons_ot3_firmware.messages import MessageDefinition
+from opentrons_ot3_firmware.utils import Int32Field
+
+from opentrons_hardware.drivers.can_bus import CanMessenger
+
+
+@pytest.mark.requires_emulator
```
Comment (sfoster1) [inline hardware/tests/firmware_integration/test_motion_constraints.py:19]: if we havent' already we should add this to the pytest ini (see https://github.com/Opentrons/opentrons/blob/edge/api/pytest.ini )

**Comment by DerekMaggio** [comment]
Is it possible that this needs a rebase after the latest merge into edge? I see a lot of OT3 refactor stuff in here

**Review by DerekMaggio — COMMENTED** [review]
Looks good to me. Will approve once we get it testing against CI

**Review by DerekMaggio — APPROVED** [review]
LGTM

**Pull request merged**